sda-download/api/s3/s3.go

```diff
@@ -196,15 +216,31 @@ func ListObjects(c *gin.Context) {
 func GetObject(c *gin.Context) {
 	log.Debugf("S3 GetObject request, context: %v", c.Params)
 
-	// Get file info for the given file path (or abort)
-	fileInfo, err := database.GetDatasetFileInfo(c.Param("dataset"), c.Param("filename")+".c4gh")
+	fileInfo, err := getFileInfo(c)
 	if err != nil {
-		if err.Error() == "sql: no rows in result set" {
-			c.AbortWithStatus(http.StatusNotFound)
-		} else {
-			c.AbortWithStatus(http.StatusInternalServerError)
```

Comment:
Shouldn't we flag the error here so the client gets a 5xx? (I don't see any need to look at the error, ISE should be good enough).

Reply:
This is good point, I'm not sure if there's any good rule either. These changes were done [quite a while ago](https://github.com/neicnordic/sda-download/commit/170f9a024449c0a81e4107bcebe67cda0ec7fd06), I do not know exactly the reason. I guess there should be tests to verify that we do get the correct error for such cases.

Reply:
Ok, created https://github.com/neicnordic/sensitive-data-archive/issues/747 for now.